### ISSUE TITLE
Prevent banned competitors from registering

### ISSUE THREAD

**Issue by viroulep** [thewca/worldcubeassociation.org#3534]
Now that we have a "Banned teams", we should definitely work on preventing these accounts from registering to competitions within their banned period.

**Comment by campos20** [comment]
We should consider also what happens if a competitor is already registrated as of the ban.

**Comment by jfly** [comment]
For the record, @pedrosino started working on this in https://github.com/thewca/worldcubeassociation.org/pull/1042, but we ended up discarding that PR. I raised a number of issues in that PR that are addressed by the new `/disciplinary` page, so I think this definitely makes sense for someone to work on again.

Any attempt to prevent banned competitors from registering is going to be "best effort". At the door registration, people registering without a WCA ID, and third party registration platforms are all things that could happen that would slip past any simple registration checks we do on the WCA website. #1037 tracks updating the results posting scripts that the Results Team uses. Updating those should handle all the cases I just described.
